packages/vite/src/node/plugins/resolve.ts

```diff
@@ -234,7 +234,7 @@ function tryFsResolve(
   if (postfixIndex < 0) {
     postfixIndex = fsPath.indexOf('#')
   }
-  if (postfixIndex > 0) {
+  if (postfixIndex > 0 && !fs.existsSync(fsPath.slice(0, postfixIndex + 1))) {
```

Comment:
Should use `accessSync` with try-catch (reason see the comment in tryResolveFile). And maybe we can extract it to a util function.

Reply:
This would only work for cases where full extensions are being used. I think the more complete fix here is:

1. Try resolving without removing postfixes first
2. If that didn't work, treat `?` and `#` as postfixes, remove them and try resolving again.